pkg/app/appdisc/factory.go

```diff
@@ -124,3 +124,13 @@ func argEqualsFlag(arg, flag string) bool {
 
 	return arg == flag
 }
+
+func argVal(args []string, flag string) string {
+	for idx, arg := range args {
+		if argEqualsFlag(arg, flag) {
+			return args[idx+1]
```

Comment:
I'm not sure, but if we pass `--passcode` as the last flag without value, wouldn't it panic with the `index out of bounds`? probably it's worth checking in here that `id < len(args)`. otherwise the value should be considered empty

Reply:
Added check.